### ISSUE TITLE
window larger than display causes SIGABRT on mac

### ISSUE THREAD

**Issue by kylemcdonald** [openframeworks/openFrameworks#5969]
in the past, creating a window larger than the display would force the window to resize to a smaller resolution (if i remember correctly). but now it causes a SIGABRT on mac. here is a minimal example:

```c++
#include "ofMain.h"
int main(){
    ofSetupOpenGL(1441, 768, OF_WINDOW);
}
```

<img width="966" alt="screen shot 2018-04-14 at 15 43 27" src="https://user-images.githubusercontent.com/157106/38773130-e3fddf88-3ffa-11e8-914f-36d2e224c238.png">

for searchability if other people have this issue, here is the error text:

```
2018-04-14 15:59:38.674253-0700 emptyExampleDebug[57932:2984689] ApplePersistence=NO
2018-04-14 15:59:38.928782-0700 emptyExampleDebug[57932:2984722] MessageTracer: Falling back to default whitelist
Assertion failed: (width >= 0), function glfwSetWindowSize, file /Users/travis/build/openframeworks/apothecary/apothecary/build/glfw/src/window.c, line 554.
```

i'm using a macbookpro retina mid-2014 which has a 1440x900 retina resolution by default. macOS 10.13.4, OF 0.10.0rc1.

**Comment by kylemcdonald** [comment]
also: if you add "High Resolution Capable" "YES" to the plist, it no longer crashes. instead, it creates a window that is the maximum size in that dimension... up to double the previous limit. for reference:

- 1024: creates 1024 non-retina window (2048 physical pixels wide)
- 1024 + hires: creates 1024 retina window (1024 physical pixels wide)
- 1441: crashes
- 1441 + hires: creates 2880 retina window (2880 physical pixels wide)
- 2800 + hires: creates 2880 retina window (2880 physical pixels wide)
- 2890 + hires: crashes

**Comment by ofTheo** [comment]
Huh. 
That does seem like a new bug. 

Before I believe it would resize it vertically but not horizontally. But definitely no crash. 

This could be a GLFW bug.

**Comment by ofTheo** [comment]
Thanks for this @kylemcdonald I am able to reproduce. Even with displays have their own spaces disabled. 

 I think the issue is due to a div by 0, which results in a negative number being passed to glfwSetWindowShape. ( see attached images ) 

I think a PR to fix this should be fairly easy. 


<img width="1397" alt="screen shot 2018-04-17 at 12 54 02 pm" src="https://user-images.githubusercontent.com/144000/38893309-cc5f74d6-423e-11e8-832f-591146a564a1.png">

<img width="632" alt="screen shot 2018-04-17 at 12 54 10 pm" src="https://user-images.githubusercontent.com/144000/38893316-d06f7a26-423e-11e8-9ee9-aa43d0917ea2.png">

**Comment by ofTheo** [comment]
@kylemcdonald this PR fixed it for me -  #5978 
Let me know if it solves it for you.   

Basically the retina check was always coming through true. because the windowW was never equal to the framebufferW if you set the window size bigger than the display. 

So now it compares the actual window W to the framebuffer W and it also stops the pixelScreenCoordScale from being zero in any other case.
